### ISSUE TITLE
Speed regression on tensorflow version > 1.4.1

### ISSUE THREAD

**Issue by jlopezpena** [tensorflow/tensorflow#19027]
We have observed some massive slowdowns in some Keras/tensorflow models with tensorflow versions newer than 1.4.1.
Not sure if the issue is with tensorflow or with the way keras creates the tensorflow models, so I am cross-posting these issue to both repos (apologies for that!).

Here is a script reproducing the issue:

## Setup
```python
from pandas import get_dummies
from sklearn.datasets import make_classification

from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.callbacks import EarlyStopping

X, y = make_classification(n_samples=10000, n_features=10, n_informative=8,
                           n_classes=5, n_clusters_per_class=1, random_state=0)

model = Sequential()
model.add(Dense(10, input_dim=10, activation='relu'))
model.add(Dropout(0.2))
model.add(Dense(5, activation='sigmoid'))
model.add(Dropout(0.1))
model.add(Dense(5, activation='softmax'))
model.compile(loss='categorical_crossentropy', optimizer='rmsprop', metrics=['acc'])

yk = get_dummies(y)
```

## Fitting

```python
model.fit(X, yk, epochs=10, batch_size=50, 
               callbacks=[EarlyStopping(monitor='loss', patience=2)])   
```

Here are some timings for the fit method:
* Tensorflow 1.4.1:  **2.91 s ± 452 ms** per loop  (obtained using ipython's `%%timeit` magic, 7 loops)
* Tensorflow 1.5.0:  CPU times: user **2min 19s**, sys: 5min 22s, total: 7min 41s Wall time: 1min 2s
* Tensorflow 1.6.0: CPU times: user **5min 5s**, sys: 12min 31s, total: 17min 36s Wall time: 2min 37s
* Tensorflow 1.7.0: CPU times: user **5min 5s**, sys: 12min 39s, total: 17min 45s Wall time: 2min 39s

So, it seems there was a massive slowdown in version 1,5, and then a further one in 1.6 (which similar speed in 1.7). All the tests are run on a conda environment with python 3.6.5 and keras 2.1.5, with the corresponding tensorflow versions all coming from the anaconda `defaults` channel.

The GPU accelerated version of keras/tensorflow (`keras-gpu` conda package) does not present the issue.

Thanks in advance!

**Comment by jlopezpena** [comment]
Could this be related to #17383 ? The magnitude of the slowdown is similar to the one reported there. Note that I see the slowdown in 1.7 as well (haven't been able to test 1.8 as there is no conda package yet)

**Comment by KonradMagnusson** [comment]
I saw about a 10% decrease in performance when going from 1.7 to 1.8.
I can't share code (NDA) but I use the same operations as you in a larger graph, and with the tf.data.Dataset API. No Keras.

**Comment by jlopezpena** [comment]
Do you also see the massive performance decrease between 1.4 and 1.5?

**Comment by drpngx** [comment]
@tfboyd @tatianashp any comment here?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jlopezpena** [comment]
The latest anaconda package (tensorflow 1.8) does not show the slowdown anymore, with timings back in line to what 1.4.1 was; so this issue can be closed.

However, it would be great if (an adequate variation of) the test above could be added as a regression test to ensure that the problem doesn't come back.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tfboyd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
